api/src/main/java/com/velocitypowered/api/proxy/server/AddressType.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2018-2023 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.proxy.server;
+
+/**
+ * Enumerates the supported Address types.
+ */
+public enum AddressType {
```

Comment:
I don't see much point in having this enum, since the addresses can already be distinguished by their class. I prefer the API from the original domain sockets commit in https://github.com/PaperMC/Velocity/commit/b00389029f4c11ad34898b5298625f4fcfc03de5

Reverting this addition would also simplify the other `InetSocketAddress -> Pair<SocketAddress, AddressType>` changes.